src/dftbp/dftbplus/parser.F90

```diff
@@ -331,24 +331,29 @@ subroutine readParserOptions(node, root, flags, implicitVersion)
     if (present(implicitVersion)) then
       call getChild(node, "ParserVersion", child, requested=.false.)
       if (associated(child)) then
-        call detailedError(child, "Cannot have both ParserVersion and InputVersion")
+        call getChildValue(child, "", inputVersion)
+        if (inputVersion /= implicitVersion) then
+          call detailedError(child, "Parser version deduced from InputVersion ("&
+          & // i2c(implicitVersion) // ") differs from version explicitely set in ParserVersion ("&
+          & // i2c(inputVersion) // ")")
+        end if
+      else
+        inputVersion = implicitVersion
+        call setChildValue(node, "ParserVersion", inputVersion, child=child)
       end if
-      inputVersion = implicitVersion
     else
-      ! Check if input needs compatibility conversion.
-      call getChildValue(node, "ParserVersion", inputVersion, parserVersion, &
-          &child=child)
+      call getChildValue(node, "ParserVersion", inputVersion, parserVersion, child=child)
     end if
+
     if (inputVersion < 1 .or. inputVersion > parserVersion) then
-      call detailedError(child, "Invalid parser version (" // i2c(inputVersion)&
-          &// ")")
-    elseif (inputVersion < minVersion) then
+      call detailedError(child, "Invalid parser version (" // i2c(inputVersion) // ")")
+    else if (inputVersion < minVersion) then
```

Comment:
Probably better to have this branch first instead of as an else if clause, as 1 <= minVersion at the moment, but might eventually change